Code-the-Dream-School/ee-prac-team3-front
# Login and Signup pages with functionality

## Purpose: 

To link front end Login and Sign up pages to backend through API.  To fetch cookie token for user through logging in.  To authenticate user using API call on Home page if cookie is present.

## Summary:

Implemented api calls for sign up and login pages allowing user to create and validate credentials. Uses cookie validation to authenticate user on Home page if cookie is present. 

## Front End Change:

App.jsx
Header.jsx
Login.jsx
SignUp.jsx
package.json{}

## Test Steps:

On home page, "Sign In" button should be visible if user is logged out.  Pressing "Sign In" should navigate user to login page.  Tester can login with email: admin@email.com and password: 1234.  Tester should be automatically navigated to the home page and "Sign In" Button should be replaced with "About" and "Quizzes" buttons in the nav bar.  Additionally, successful login should result in cookie token visible via developer tools.  On redirect to the home page, user data should be stored locally in authContextProvider.  Name Data should be passed to the Quiz.jsx component and used to create a USER ICON with two letters, the capital first letter of the user's first name and the capital first letter of the users last name.

## If Encountering Eslint and Prettier Errors 

try to install prettier, eslint prettier plugin, and prettier eslint config

npm install --save-dev prettier

npm install --save-dev eslint-plugin-prettier

npm install --save-dev eslint-config-prettier

## For Backend Functionality

Backend in Dev branch is currently missing middleware. I implemented the following changes to make it work.

in src/routes/mainRouter.js  replace the current /login route with the following

router.route("/login").post(loginDataValidate, logIn).get(authenticateUser, (req, res) => {
    // Handler for GET request after successful authentication
    res.json({
      success: true,
      message: "User successfully authenticated",
      user: req.user
    });
  });

Backend Dev branch currently requires username to signup, but there is no username field in the signup sheet.  To remove username requirement on the back-end, remove the "required" key in the username portion of the userschema.js and remove username requirements from signupDataValidate.js .
